### ISSUE TITLE
Advice using pubchem with commonJS modules

### ISSUE THREAD

**Issue by mikemaster98** [cheminfo/pubchem#21]
I am completely unexperienced with javascript, so it is really difficult for me to code a webscraper in NodeJS to acquire molecule information from PubChem. I would therefore love to be able to use this pubchem instead. However, (sorry if I am misunderstanding), my NodeJS system is based on commonJS modules, so I use the require() function. 'pubchem' seems ES6 based and so I have to use import. I have managed to bypass the "import" error by using a dynamic import ('let Compound = await import('pubchem') ), but it now gives an error that "cache is undefined". 

Simply put, how can I use your module without having to convert all mine to ES6? Also, what is the argument that is passed on to the code described in the front page?

**Comment by kjappelbaum** [comment]
you can look into using the version from the CDN (https://www.lactame.com/lib/pubchem/0.1.0/pubchem.js) which is transpiled (or you can [transpile yourself](https://babeljs.io/)). 

Note though that the API is still evolving and will probably undergo many more changes as we ourselves work with it.

**Comment by mikemaster98** [comment]
The transpiled version seems to bring its own errors. Somehow I cannot seem to fix the error 
> "Error: unable to locate global object"

I have added the error and the corresponding code. typeof exports and module is indeed object. 
globalThis is also not undefined. 

As I am dealing with Node.js v12 by the way. globalThis should provide the global object in Node, so I don't know where it is going wrong. 



```
    at getGlobal (.......\pubchem3.js:39:19)
    at ......\pubchem3.js:42:22
    at createCommonjsModule (.......\pubchem3.js:18:16)
    at ......\pubchem3.js:25:19
    at .......\pubchem3.js:8:68
    at .......\pubchem3.js:11:2)

```

```
(function (global, factory) {
         /* line 8 --> */ typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports) :
        typeof define === 'function' && define.amd ? define(['exports'], factory) :
            (global = typeof globalThis !== 'undefined' ? globalThis : global || self, factory(global.PubChem = {}));
}(this, (function (exports) {       /*<-- line 11*/
    'use strict';
    var commonjsGlobal = typeof globalThis !== 'undefined' ? globalThis : typeof global !== 'undefined' ? global : typeof global !== 'undefined' ? global : typeof self !== 'undefined' ? self : {};
    function createCommonjsModule(fn) {
        var module = { exports: {} };
        /* line 18 --> */   return fn(module, module.exports), module.exports;
    }
    function commonjsRequire(target) {
        throw new Error('Could not dynamically require "' + target + '". Please configure the dynamicRequireTargets option of @rollup/plugin-commonjs appropriately for this require call to behave properly.');
    }
    /* line 25 --> */     var browser = createCommonjsModule(function (module, exports) {
        var getGlobal = function () {
            // the only reliable means to get the global object is
            // `Function('return this')()`
            // However, this causes CSP violations in Chrome apps.
            if (typeof self !== 'undefined') {
                return self;
            }
            if (typeof global !== 'undefined') {
                return global;
            }
                /* line 39 --> */    throw new Error('unable to locate global object');
        };
        /* line 42 --> */        var global = getGlobal();
        module.exports = exports = global.fetch; // Needed for TypeScript and Webpack.
        if (global.fetch) {
            exports.default = global.fetch.bind(global);
        }
```
